tools/cli/internal/openapi/filter/hidden_envs.go

```diff
@@ -109,19 +123,17 @@ func (f *HiddenEnvsFilter) removeResponseIfHiddenForEnv(operation *openapi3.Oper
 		if response.Value == nil || response.Value.Content == nil {
 			continue
 		}
-		for _, contentType := range response.Value.Content {
-			f.removeContentIfHiddenForEnv(contentType)
-		}
-	}
-}
 
-func (f *HiddenEnvsFilter) removeContentIfHiddenForEnv(contentType *openapi3.MediaType) {
-	if isContentTypeHiddenForEnv := isContentTypeHiddenForEnv(contentType, f.metadata.targetEnv); isContentTypeHiddenForEnv {
-		log.Printf("Removing contentType: %q because is hidden for target env: %q", contentType.Schema.Ref, f.metadata.targetEnv)
-		contentType.Schema = nil // Remove ContentType if it is hidden for the target environment
```

Comment:
`contentType.Schema = nil // Remove ContentType if it is hidden for the target environment` This piece of code was removing the schema instead of the ContentType

Reply:
The main reason was that we need to pass the `k` and `response.Value.Content` as we need them to delete the content.